gateway-service/src/main/java/org/zowe/apiml/gateway/filters/pre/TomcatFilter.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+package org.zowe.apiml.gateway.filters.pre;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Component;
+import org.zowe.apiml.message.core.Message;
+import org.zowe.apiml.message.core.MessageService;
+import org.zowe.apiml.message.log.ApimlLogger;
+import org.zowe.apiml.product.logging.annotations.InjectApimlLogger;
+
+import javax.servlet.*;
+import javax.servlet.annotation.WebFilter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.net.URLDecoder;
+
+@Component
+@WebFilter
+@RequiredArgsConstructor
+public class TomcatFilter implements Filter {
+
+    private final MessageService messageService;
+    private final ObjectMapper mapper;
+
+    @Value("${apiml.service.allowEncodedSlashes:#{false}}")
+    private Boolean allowEncodedSlashes;
```

Comment:
May I ask why reference type? It requires to check whether is null:
```
boolean allowEncoded = allowEncodedSlashes != null && allowEncodedSlashes;
```

Reply:
I did it because of unit tests, where we need to reset this field, and for this porpose I used 'ReflectionTestUtils.setField', where the value of field is Object. I forgot about autoboxing...